domovoi.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import shlex
+import sys
+import time
+
+import settings
+
+from solar_car_process import SolarCarProcess
+
+
+class Domovoi:
+        
+    def run(self, processes_file):
+        with open(processes_file) as file:
+            solar_car_processes = [SolarCarProcess(shlex.split(path)) for path in file.read().splitlines()]
+        
+        # Check processes and respond accordingly
+        while True:
+            for solar_car_process in solar_car_processes:
+                if solar_car_process.check_status():
+                    if solar_car_process.timesRestarted == settings.MAX_RESTART:
+                        logging.critical("%s reached %d restart(s) with signal %d",
+                            os.path.basename(solar_car_process.path[0]), settings.MAX_RESTART, solar_car_process.process.returncode)
+                        solar_car_processes.remove(solar_car_process)
+                    else:
+                        solar_car_process.start()
```

Comment:
If the problem crashed in the first place, it is probably likely it will have future problems starting. We should check for any exceptions raised from `subprocess.Popen` and check for other errors if possible.